### PR TITLE
Fix kfp pipelines testing in github workflow.

### PR DESCRIPTION
**Pull request by revit13**
## Why are these changes needed?

Fix kfp pipelines testing in github workflow.

### CODE REVIEW

File: .github/workflows/test-kfp-transform.template [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - "dev"
+            - "releases/**"
+        tags:
+            - "*"
+        paths:
+            - "@TARGET_TRANSFORM_DIR@/**"
```
Comment (daw3rd) [inline .github/workflows/test-kfp-transform.template:15]: I thought we wanted to make these run ONLY when @TARGET_TRANSFORM_DIR@/kfp_ray changed?

File: .github/workflows/test-code-inputcode2parquet.yml
Comment (daw3rd) [inline .github/workflows/test-code-inputcode2parquet.yml:1]: why is this file being added?  I don't see the code/inputcode2parquet transform in dev

**Review by daw3rd — CHANGES_REQUESTED** [review]
Also, because we now have 2 workflows per transform maybe we shoulld change the naming convention.  For example, universal-noop-test-kfp.yml and universal-noop-test.yml.?

File: .github/workflows/test1.yml
Comment (daw3rd) [inline .github/workflows/test1.yml:1]: what is this file doing?  It is not clear.  Maybe you did not mean to push it?

**Review by daw3rd — CHANGES_REQUESTED** [review]

File: .github/workflows/test-kfp-transform.template (hunk 1 above)
Comment (revit13) [reply .github/workflows/test-kfp-transform.template:15]: I think we should run kfp when a specific transform changes to make sure the transform parameters and pipeline parameters match after a change.

File: .github/workflows/test1.yml
Comment (revit13) [reply .github/workflows/test1.yml:1]: Yes, thanks. will revert it.

File: .github/workflows/Makefile [hunk 2]
```diff
@@ -7,20 +7,41 @@ CODE_TRANSFORMS=code2parquet code_quality header_cleanser malware proglang_selec
...
 
 
+# A list that holds transforms that should not be tested with KFP
+KFP_BLACK_LIST="doc_chunk,pdf2parquet,pii_redactor"
+
 transform-tests:
-	$(MAKE) TRANSFORM_SUBDIR=universal .transform-tests
-	$(MAKE) TRANSFORM_SUBDIR=language .transform-tests
-	$(MAKE) TRANSFORM_SUBDIR=code .transform-tests
+	$(MAKE) TRANSFORM_SUBDIR=universal .transform-tests .transform-kfp-tests
```
Comment (roytman) [inline .github/workflows/Makefile]: should it be `$(MAKE) TRANSFORM_SUBDIR=universal .transform-tests ` ?

File: data-processing-lib/ray/README.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Data Processing Library
-This provides a python framework for developing _transforms_
+This provides a python framework for developing _transforms_99
```
Comment (roytman) [inline data-processing-lib/ray/README.md]: what is the `99` suffix?

File: kfp/kfp_ray_components/README.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ Once the components are implemented we also implement their interfaces as a comp
 
 * The component’s inputs and outputs.
 * The container image that your component’s code runs in, the command to use to run your component’s code, and the 
-command-line arguments to pass to your component’s code.
+command-line arguments to pass to your component’s code99.
```
Comment (roytman) [inline kfp/kfp_ray_components/README.md]: as above

File: data-processing-lib/ray/README.md (hunk 3 above)
Comment (revit13) [reply data-processing-lib/ray/README.md]: left over from testing sorry about that. Fixed

File: kfp/kfp_ray_components/README.md (hunk 4 above)
Comment (revit13) [reply kfp/kfp_ray_components/README.md]: left over from testing sorry about that. Fixed

File: .github/workflows/Makefile (hunk 2 above)
Comment (revit13) [reply .github/workflows/Makefile]: Fixed. Thanks

**Review by roytman — APPROVED** [review]
LGTM

File: .github/workflows/test-kfp-transform.template (hunk 1 above)
Comment (daw3rd) [reply .github/workflows/test-kfp-transform.template:15]: if that is the case, then we don't think we need a special template and we can reinstate `test-src` target in kfp_ray/Makefile.

Comment (daw3rd) [reply .github/workflows/test-kfp-transform.template:15]: ok

File: .github/workflows/test-kfp-transform.template [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            - "!**/doc/**"
+            - "!**/images/**"
+            - "!**.gitignore"
+    pull_request:
+        branches:
+            - "dev"
+            - "releases/**"
+        paths:
+            - "@TARGET_TRANSFORM_DIR@/**"
+            - "!data-processing-lib/**" # This is tested in separate workflow
```
Comment (daw3rd) [inline .github/workflows/test-kfp-transform.template:28]: I don't believe. you needs this NOTs on data-processing-lib or kfp

File: .github/workflows/test-kfp.yml [hunk 6]
```diff
@@ -20,6 +28,15 @@ on:
             - "releases/**"
         paths:
             - "kfp/**"
+            - "!kfp/**/test/**"
+            - "!kfp/**.md"
+            - "!kfp/**/doc/**"
+            - "data-processing-lib/**"
+            - "!data-processing-lib/**/test/**"
```
Comment (daw3rd) [inline .github/workflows/test-kfp.yml:35]: Should not need this NOTs on data-processing-lib

**Review by daw3rd — CHANGES_REQUESTED** [review]

File: .github/workflows/test-kfp.yml (hunk 6 above)
Comment (daw3rd) [reply .github/workflows/test-kfp.yml:35]: oops my bad. I did not see the inclusion on data-processing-lib/**

File: .github/workflows/README.md [hunk 7]
```diff
@@ -1,20 +1,23 @@
...
 
 #### Goals
 1. Run only tests for a given transform when only the transform changes.
 Includes python, ray, spark and kfp_ray as available.  
 2. When the core dpk lib components files changes, test all transforms
-3. When the shared kfp components changes, test a randomly selected transform test
-   (We would like to avoid running all transform kfp tests in one PR)
+3. When the shared kfp components changes or core dpk lib components files changes, 
+   test a randomly selected transform test. Otherwise run kfp test for the changed transforms.
 4. Extra credit: If .md or other non-code changes are made, run no tests.
```
Comment (daw3rd) [inline .github/workflows/README.md]: i think this number 4 can be removed.

**Review by daw3rd — APPROVED** [review]

**Pull request merged**